lib/helpers.js

```diff
@@ -42,20 +42,48 @@ const format = (event, options = { ignoreSelection: false }) => {
   const selectedText = editor.getSelectedText();
   const isTransformingFile = options.ignoreSelection || !selectedText;
 
-  const textToTransform = isTransformingFile ? editor.getText() : selectedText;
+  const currentScope = getCurrentScope();
+  const isEmbedded = ['text.html.vue', 'text.html.basic'].includes(currentScope);
+  if (isEmbedded) {
+    const re = /<script\b[^>]*>([\s\S]*?)(?=<\/script>)/gi
+    editor.scan(re, iter => {
+      // only transform first match
+      iter.stop();
+
+      // Create new range with start row advanced by 1,
+      // since we cannot use look-behind on variable-length starting
+      // <script ...> tag
+      const { start, end } = iter.range;
+      const range = new iter.range.constructor(
+        [start.row + 1, start.column],
+        end
+      );
+
+      // Transform range
+      const textToTransform = editor.getTextInBufferRange(range);
+      const transformed = executePrettier(textToTransform, getFormatOptions());
+      if (!transformed) {
+        return;
+      }
+      editor.setTextInBufferRange(range, transformed);
+      editor.setCursorScreenPosition(cursorPositionPriorToFormat);
+    })
+  } else {
+    const textToTransform = isTransformingFile ? editor.getText() : selectedText;
 
-  const transformed = executePrettier(textToTransform, getFormatOptions());
-  if (!transformed) {
-    return;
-  }
+    const transformed = executePrettier(textToTransform, getFormatOptions());
+    if (!transformed) {
+      return;
+    }
```

Comment:
it seems like there is some room to DRY this up, and the method is getting super-long now. Maybe we should extract method into something like `getTextToTransform(editor, isEmbedded)` or something?